diff --git a/nixos/modules/services/mail/maddy.nix b/nixos/modules/services/mail/maddy.nix
index 0bd3a0c2bd99b..cf17a862a32c2 100644
--- a/nixos/modules/services/mail/maddy.nix
+++ b/nixos/modules/services/mail/maddy.nix
@@ -350,11 +350,13 @@ in
       };
 
       secrets = lib.mkOption {
-        type = with lib.types; listOf path;
+        type = with lib.types; listOf (either str path);
         description = ''
           A list of files containing the various secrets. Should be in the format
           expected by systemd's `EnvironmentFile` directory. Secrets can be
           referenced in the format `{env:VAR}`.
+
+          Paths can be prefixed with `-` to ignore errors if the file does not exist.
         '';
         default = [ ];
       };
