diff --git a/nixos/modules/services/desktop-managers/gnome.nix b/nixos/modules/services/desktop-managers/gnome.nix
index 1d850f1f26e5a..6dd4ceec4370e 100644
--- a/nixos/modules/services/desktop-managers/gnome.nix
+++ b/nixos/modules/services/desktop-managers/gnome.nix
@@ -400,6 +400,15 @@ in
         pkgs.xdg-user-dirs-gtk # Used to create the default bookmarks
       ];
 
+      # Restarting this unit terminates the active GNOME session.
+      systemd.user.services.gnome-session-monitor = {
+        restartIfChanged = false;
+        overrideStrategy = "asDropin";
+        # No need to add the NixOS default Environment="Path=coreutils:...",
+        # to the gnome-session-monitor service.
+        enableDefaultPath = false;
+      };
+
       services.udev.packages = [
         # Force enable KMS modifiers for devices that require them.
         # https://gitlab.gnome.org/GNOME/pkgs.mutter/-/merge_requests/1443
