diff --git a/nixos/modules/services/networking/murmur.nix b/nixos/modules/services/networking/murmur.nix
index 8f72e3c52b849..6b92a61c7be6c 100644
--- a/nixos/modules/services/networking/murmur.nix
+++ b/nixos/modules/services/networking/murmur.nix
@@ -101,7 +101,7 @@ in
       };
 
       autobanAttempts = lib.mkOption {
-        type = lib.types.int;
+        type = lib.types.ints.unsigned;
         default = 10;
         description = ''
           Number of attempts a client is allowed to make in
@@ -111,7 +111,7 @@ in
       };
 
       autobanTimeframe = lib.mkOption {
-        type = lib.types.int;
+        type = lib.types.ints.unsigned;
         default = 120;
         description = ''
           Timeframe in which a client can connect without being banned
@@ -120,7 +120,7 @@ in
       };
 
       autobanTime = lib.mkOption {
-        type = lib.types.int;
+        type = lib.types.ints.unsigned;
         default = 300;
         description = "The amount of time an IP ban lasts (in seconds).";
       };
@@ -154,7 +154,7 @@ in
       };
 
       bandwidth = lib.mkOption {
-        type = lib.types.int;
+        type = lib.types.ints.unsigned;
         default = 72000;
         description = ''
           Maximum bandwidth (in bits per second) that clients may send
@@ -163,19 +163,19 @@ in
       };
 
       users = lib.mkOption {
-        type = lib.types.int;
+        type = lib.types.ints.unsigned;
         default = 100;
         description = "Maximum number of concurrent clients allowed.";
       };
 
       textMsgLength = lib.mkOption {
-        type = lib.types.int;
+        type = lib.types.ints.unsigned;
         default = 5000;
         description = "Max length of text messages. Set 0 for no limit.";
       };
 
       imgMsgLength = lib.mkOption {
-        type = lib.types.int;
+        type = lib.types.ints.unsigned;
         default = 131072;
         description = "Max length of image messages. Set 0 for no limit.";
       };
