diff --git a/nixos/modules/services/x11/desktop-managers/phosh.nix b/nixos/modules/services/x11/desktop-managers/phosh.nix
index 36d8a63318078..3a76f92c91e8f 100644
--- a/nixos/modules/services/x11/desktop-managers/phosh.nix
+++ b/nixos/modules/services/x11/desktop-managers/phosh.nix
@@ -168,6 +168,7 @@ in
   };
 
   config = lib.mkIf cfg.enable {
+    services.displayManager.enable = true;
     # Inspired by https://gitlab.gnome.org/World/Phosh/phosh/-/blob/main/data/phosh.service
     # Parts taken from nixos/modules/services/wayland/cage.nix
     systemd.services.phosh = {
