diff --git a/nixos/modules/services/web-servers/phpfpm/default.nix b/nixos/modules/services/web-servers/phpfpm/default.nix
index 43d394ad39f46..75177c7f96871 100644
--- a/nixos/modules/services/web-servers/phpfpm/default.nix
+++ b/nixos/modules/services/web-servers/phpfpm/default.nix
@@ -47,12 +47,14 @@ let
     poolOpts:
     pkgs.runCommand "php.ini"
       {
-        inherit (poolOpts) phpPackage phpOptions;
+        inherit (poolOpts) phpOptions;
         preferLocalBuild = true;
-        passAsFile = [ "phpOptions" ];
+        __structuredAttrs = true;
       }
       ''
-        cat ${poolOpts.phpPackage}/etc/php.ini $phpOptionsPath > $out
+        ( cat ${poolOpts.phpPackage}/etc/php.ini
+          printf "%s" "$phpOptions"
+        ) > $out
       '';
 
   poolOpts =
